acceptance/ui/features/services.feature

```diff
@@ -29,4 +29,181 @@ Feature: Service Management
+    And "table://services/childServiceWithVHost/svc_default" should be selected for Service Endpoint
+    And "table://services/childServiceWithVHost/protocol_default" should be selected for Protocol
+
+  @services_endpoint
+  Scenario: Public Endpoint dialog - no port given
+    When I view the details of "table://services/topService/name" in the "Applications" table
+      And I view the details of "table://services/childServiceWithVHost/name" in the "Services" table
+      And I click "Add Public Endpoint"
+      And I should see the Add Public Endpoint dialog
+      And I click "Add and Restart Service"
+    Then I should see "Missing port"
+
+  @services_endpoint
+  Scenario: Public Endpoint dialog - bad port given
+    When I view the details of "table://services/topService/name" in the "Applications" table
+      And I view the details of "table://services/childServiceWithVHost/name" in the "Services" table
+      And I click "Add Public Endpoint"
+      And I should see the Add Public Endpoint dialog
+      And I fill in Port "99999"
+      And I click "Add and Restart Service"
+    Then I should see "Port must be between 1 and 65536"
+
+  @services_endpoint
+  Scenario: Public Endpoint dialog - bad host given
+    When I view the details of "table://services/topService/name" in the "Applications" table
+      And I view the details of "table://services/childServiceWithVHost/name" in the "Services" table
+      And I click "Add Public Endpoint"
+      And I should see the Add Public Endpoint dialog
+      And I fill in Host "a.b.c.d"
+      And I fill in Port "23456"
+      And I click "Add and Restart Service"
+    Then I should see "no such host"
+
+  @services_endpoint
+  Scenario: Public Endpoint dialog - add https
+    When I view the details of "table://services/topService/name" in the "Applications" table
+      And I view the details of "table://services/childServiceWithVHost/name" in the "Services" table
+      And I click "Add Public Endpoint"
+      And I should see the Add Public Endpoint dialog
+      And I select Service "table://services/childServiceWithVHost/name" - "table://services/childServiceWithVHost/endpoint_https"
+      And I fill in Host "table://services/childServiceWithVHost/host"
+      And I fill in Port "table://services/childServiceWithVHost/port_https"
+      And I select Protocol "table://services/childServiceWithVHost/protocol_https"
+      And I click "Add and Restart Service"
+    Then I should see "table://services/childServiceWithVHost/name"
+      And I should see "Public Endpoints"
+      And Endpoint details should be service:"table://services/childServiceWithVHost/name" endpoint:"table://services/childServiceWithVHost/endpoint_https" type:"port" protocol:"table://services/childServiceWithVHost/protocol_https_display" host:"table://services/childServiceWithVHost/host" port:"table://services/childServiceWithVHost/port_https"
```

Comment:
I thought I made this comment in the first round, but I guess I didn't click Save or something. Anyway, please split this step into separate steps so that only one value value is verified in each step. For instance something like:
```
And Endpoint details should have service: "table://services/childServiceWithVHost/name"
And Endpoint details should have endpoint: "table://services/childServiceWithVHost/endpoint_https"
...
```
This makes it easier to combine steps in future tests, and it also makes it easier to see what is wrong when a single value is incorrecct.